Core/Models/ImapClientModel.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System.Collections.Generic;
+
+using MailKit.Net.Imap;
+using MailKit;
+using System;
+
+namespace Core.Models
+{
+    public class ImapClientModel
+    {
+        public string ActiveFolder
+        {
+            get { return _activeFolder; }
+            set
+            {
+                if (value != _activeFolder)
+                {
+                    // active folder changed, so list of headers must be downloaded for a new folder
+                    _headers = null;
+                }
+                _activeFolder = value;
+            }
+        }
+
+        public List<IMailFolder> Folders
+        {
+            get
+            {
+                if (_folders == null)
+                {
+                    _folders = new List<IMailFolder>();
+                    _downloadFolders();
+                }
+                return _folders;
+            }
+        }
+
+        // list of headers in active folder
+        public IList<IMessageSummary> Headers
+        {
+            get
+            {
+                if (_headers == null)
+                {
+                    _downloadHeaders();
+                }
+                return _headers;
+            }
+        }
+
+        private readonly string _login;
+        private readonly string _password;
+        private readonly string _host;
+        private readonly int _port;
+        private readonly bool _useSsl;
+
+        private ImapClient _client;
+        private string _activeFolder;
+        private List<IMailFolder> _folders;
+        private IList<IMessageSummary> _headers;
+        private Dictionary<Tuple<string, UniqueId>, MimeKit.MimeMessage> _messages;
+        private bool _connected;
+
+        public ImapClientModel(string login, string password, string host, int port, bool useSsl)
+        {
+            _login = login;
+            _password = password;
+            _host = host;
+            _port = port;
+            _useSsl = useSsl;
+            _connected = false;
+            _messages = new Dictionary<Tuple<string, UniqueId>, MimeKit.MimeMessage>();
+        }
+
+        public void Connect()
+        {
+            _client = new ImapClient();
+            _client.ServerCertificateValidationCallback = (s, c, h, e) => true;
```

Comment:
Is it safe to ignore SSL certificate?

Reply:
I would leave it for now and implement it later